### PR TITLE
DFPL-2167  Playwright script for Respondents' Details

### PR DESCRIPTION
**Pull request by Braimah101**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-2167

### Change description ###
As part of the FPL, we need to create a  Hearing Urgency to the new Playwright framework. This task involves migrating the test suite and updating any necessary configurations and tests as required.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: playwright-e2e/pages/respondents-details.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+async RespondentDetailsNeeded() {
+  await this.page.getByRole('heading', { name: 'Add information about the' }).isVisible();
+  await this.page.getByRole('link', { name: 'Respondents\' details' }).click();
+  await this.page.getByLabel('*First name (Optional)').fill('Asa');
+  await this.page.getByLabel('*Last name (Optional)').click();
+  await this.page.getByLabel('*Last name (Optional)').fill('Thierry');
+  await this.page.getByText('Date of birth (Optional)').click();
+  await this.page.getByLabel('Day').isVisible();
+  await this.page.pause();
```
Comment (GauravChawlaMOJ) [inline playwright-e2e/pages/respondents-details.ts:37]: pause has to be removed for test to run

File: playwright-e2e/pages/respondents-details.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  await this.page.getByLabel('Day').fill('20');
+  await this.page.getByLabel('Month').click();
```
Comment (GauravChawlaMOJ) [inline playwright-e2e/pages/respondents-details.ts:39]: Day, month, year, gender etc. can be defined just once and reused

File: playwright-e2e/pages/risk-and-harm-to-children.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { type Page, type Locator, expect } from "@playwright/test";
 
-export class RiskAndHarmToChildren { 
+export class RiskAndHarmToChildren {
+    respondentsDetailsSmokeTest() {
+      throw new Error("Method not implemented.");
```
Comment (GauravChawlaMOJ) [inline playwright-e2e/pages/risk-and-harm-to-children.ts:5]: is this method needed here?

File: playwright-e2e/pages/start-application.ts [hunk 4]
```diff
@@ -12,6 +17,10 @@ export class StartApplication {
   readonly hearingUrgencyHeader: Locator;
   readonly groundsForTheApplicationHeading: Locator;
   readonly groundsForTheApplicationHasBeenUpdatedFinished: Locator;
+  readonly
```
Comment (GauravChawlaMOJ) [inline playwright-e2e/pages/start-application.ts:20]: this might be mistakenly added, remove readonly from line 20?

**Review by GauravChawlaMOJ — CHANGES_REQUESTED** [review]

File: playwright-e2e/pages/respondents-details.ts (hunk 2 above)
Comment (Braimah101) [reply playwright-e2e/pages/respondents-details.ts]: True

File: playwright-e2e/pages/start-application.ts (hunk 4 above)
Comment (Braimah101) [reply playwright-e2e/pages/start-application.ts]: true

File: playwright-e2e/pages/risk-and-harm-to-children.ts (hunk 3 above)
Comment (Braimah101) [reply playwright-e2e/pages/risk-and-harm-to-children.ts]: True

File: playwright-e2e/tests/smoke-test.spec.ts [hunk 5]
```diff
@@ -27,39 +30,72 @@ test("Smoke Test @smoke-test", async ({
...
-  await hearingUrgency.whatTypeOfHearingDoYouNeed("Standard case management");
-  await hearingUrgency.giveReasonTextBoxFill();
-  await hearingUrgency.withoutNoticeHearing("No");
-  await hearingUrgency.needAHearingWithReducedNoise("No");
-  await hearingUrgency.respondentsAwareOfProceedings("No");
-  await hearingUrgency.continueButton.click();
-  await hearingUrgency.checkYourAnswers.isVisible();
-  await hearingUrgency.saveAndContinueButton.click();
-  await startApplication.addApplicationDetailsHeading.isVisible();
+  // await startApplication.hearingUrgencyLink.isVisible();
```
Comment (Iphiclus) [inline playwright-e2e/tests/smoke-test.spec.ts:39]: Hi @Braimah101 is there a reason for these lines to be commented out of the smoke test?

File: playwright-e2e/tests/create-case.spec.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import { test, expect } from "@playwright/test";
```
Comment (Iphiclus) [inline playwright-e2e/tests/create-case.spec.ts:1]: Hi @Braimah101 is there a reason this test has been added back in? we did remove it, it is also currently skipped.

**Review by Iphiclus — CHANGES_REQUESTED** [review]

File: playwright-e2e/tests/smoke-test.spec.ts (hunk 5 above)
Comment (Braimah101) [reply playwright-e2e/tests/smoke-test.spec.ts:39]: True

**Pull request closed without merging**